### PR TITLE
Fix vps

### PR DESCRIPTION
**Pull request by aljones15**
Adding working VP tests to suite:

1. Sign VPs  with a challenge and VCs with a proof.
2. Remove empty `verifiableCredential` properties from VPs.
3. Switch from using a bs58 secret to a keyDoc with did:key ids set.

### CODE REVIEW

File: tests/10-vcdm2.js [hunk 1]
```diff
@@ -97,20 +98,25 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
       };
       const result = await post(vpVerifier, body);
-      if(result.errors.length) {
+      if(result?.errors?.length) {
         throw result.errors[0];
       }
       return result;
     }
 
+    const challenge = 'z19pnWMzs2Ub452WW9BLBJP9h';
```
Comment (msporny) [inline tests/10-vcdm2.js:113]: Generate a new challenge each time to ensure implementations are not hard coding the challenge.

File: tests/10-vcdm2.js [hunk 2]
```diff
@@ -97,20 +98,25 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
+    const proveOptions = {challenge};
+    const verifyPresentationOptions = {
+      checks: ['proof'],
```
Comment (msporny) [inline tests/10-vcdm2.js:116]: It feels wrong to not at least check the proof when verifying. What happens if `proof` isn't listed in the checks, or the checks are empty? If the answer is: success, then we need to discuss.

**Review by msporny — CHANGES_REQUESTED** [review]
Only one change request. 

The rest are comments that we'll need to resolve outside of this PR.

File: tests/10-vcdm2.js (hunk 1 above)
Comment (aljones15) [reply tests/10-vcdm2.js]: I used `Date.now()` as the challenge: https://github.com/digitalbazaar/vc-data-model-2-test-suite/pull/26/commits/a6158ad9e20d07bccb3070cc728b222d1be0e15f

As a String not a number.

File: tests/10-vcdm2.js [hunk 3]
```diff
@@ -156,13 +161,12 @@ describe('Verifiable Credentials Data Model v2.0', function() {
         const vc = await issue(require('./input/credential-ok.json'));
         assert(Array.isArray(vc['@context']));
         assert.strictEqual(vc['@context'][0], baseContextUrl);
-        //FIXME reimplement this once VP creation via VC-API
-        //has stabilized
+        //FIXME reimplement this once signed VP creation via VC-API
+        //has been finalized
```
[the remark this answers is not in the document]
Comment (aljones15) [reply tests/10-vcdm2.js:165]: meeting attended and we will be implementing `presentation/create` in the future.

File: tests/10-vcdm2.js (hunk 2 above)
Comment (aljones15) [reply tests/10-vcdm2.js:116]: This was discussed internally and was considered to be ok.

File: tests/10-vcdm2.js (hunk 1 above)
Comment (msporny) [reply tests/10-vcdm2.js]: No, don't use Date.now() -- use a 128-bit random number and base64url encode the value w/ a multibase prefix of 'u'.

File: tests/10-vcdm2.js (hunk 2 above)
Comment (msporny) [reply tests/10-vcdm2.js:116]: What is happening internally is of no consequence to the other implementers. This is something that other implementers might feel that they have to implement, so we have to think more holistically about if this is required for other implementers. At present, I guess the answer is "no, it's not required"... we can leave it for now, but should document that we're looking for implementer feedback on the option.

**Review by msporny — CHANGES_REQUESTED** [review]
Don't use Date.now() -- that's not an entropy source with enough randomness -- we don't want people copying that approach -- use a 128-bit random number and base64url encode the value w/ a multibase prefix of 'u'.

File: tests/10-vcdm2.js (hunk 1 above)
Comment (aljones15) [reply tests/10-vcdm2.js]: Using a base64 encoded 128 byte number here: https://github.com/digitalbazaar/vc-data-model-2-test-suite/pull/26/commits/137554f8c18ae1a3e861ee74af0aa9c231f1345c

File: tests/10-vcdm2.js (hunk 2 above)
Comment (aljones15) [reply tests/10-vcdm2.js:117]: We are checking the proof when verifying if that helps to assuage any concerns and proof if the check we are requiring currently for VPs.

File: tests/10-vcdm2.js (hunk 1 above)
Comment (aljones15) [reply tests/10-vcdm2.js]: added the `u` multibase prefix here: https://github.com/digitalbazaar/vc-data-model-2-test-suite/pull/26/commits/5cd3784f7eaa087ad38f82e3fede0b439faa8686

**Review by msporny — APPROVED** [review]

File: tests/10-vcdm2.js [hunk 4]
```diff
@@ -97,19 +98,25 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
-        }
+        options
       };
       const result = await post(vpVerifier, body);
       if(result?.errors?.length) {
         throw result.errors[0];
       }
       return result;
     }
+    // use base64 encoded 128 byte number as the challenge
```
Comment (msporny) [inline tests/10-vcdm2.js:112]: 128-bit random number, not 128-byte random number, :)

**Review by msporny — CHANGES_REQUESTED** [review]
Random number generation looks incorrect. 

```js
const { randomFillSync } = await import('node:crypto');
const buf = Buffer.alloc(16); // 128 bits
console.log('u' + randomFillSync(buf).toString('base64url'));
```

**Review by msporny — APPROVED** [review]

**Pull request merged**